src/main/java/pages/yandex/HomePage.java

```diff
@@ -1,29 +1,33 @@
 package pages.yandex;
 
+import helperClasses.SingletonBrowser;
 import lombok.Getter;
 import lombok.Setter;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
-import projectVeriables.ProjectVariables;
 
 @Getter
 @Setter
 public class HomePage {
-    final WebDriver driver;
-
-    public HomePage(WebDriver driver) {
-        this.driver = driver;
-        this.driver.get(ProjectVariables.MAIN_URL);
-
-    }
+    private final WebDriver driver;
 
     private static final By HOME_PAGE_TITLE = By.xpath("//div[@class=\"HeadBanner-Title\"]");
     private static final By CREATE_ACCOUNT_BUTTON = By.xpath("//div[@class='HeadBanner-ButtonsWrapper']/a/span[contains(text(),'Create an account')]");
     private static final By LOG_IN_BUTTON = By.className("HeadBanner-Button-Enter");
     private static final By ACCEPT_COOKIES = By.xpath("//button[@data-id=\"button-all\"]");
 
+    public HomePage() {
+        this.driver = SingletonBrowser.getInstance().getDriver();
+    }
+
     public LogInPage clickOnLogInButton() {
         driver.findElement(LOG_IN_BUTTON).click();
-        return new LogInPage(driver);
+        return new LogInPage();
+    }
+
+    public boolean verifyHomePage(){
+        driver.findElement(CREATE_ACCOUNT_BUTTON).isDisplayed();
+        driver.findElement(LOG_IN_BUTTON).isDisplayed();
+        return true;
```

Comment:
Let me know, when this method will return "false" and catch potential bug

Reply:
hmm I have tested if locator is wrong and it works. 
Probably if element is displayed but DISABLED and cannot be clicked or overlapping some other html element

Reply:
I don't know what are you testing, but this method always returns "true" as I see in code and this is very weird.